sqlmesh/core/context.py

```diff
@@ -306,6 +306,14 @@ def __init__(
         self.users = list({user.username: user for user in self.users}.values())
         self._register_notification_targets()
 
+        if (
+            self.config.environment_catalog_mapping
+            and not self.engine_adapter.CATALOG_SUPPORT.is_multi_catalog_supported
```

Comment:
Should we also check whether the cross-catalog querying is supported?

Reply:
That is what this is checking

Reply:
Ah, I thought this only checked whether multiple catalogs are supported at all. Got confused